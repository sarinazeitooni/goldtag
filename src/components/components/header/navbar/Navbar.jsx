import React from 'react';
import './style/navbar.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import DropDown from "./dropdown/Dropdown";
import texts from './texts/texts';
import {faStopwatch, faIdCard, faUser, faUserNurse}
from "@fortawesome/free-solid-svg-icons";
function Navbar(){
    return(
        <div id="navbar" style={{backgroundColor: 'hsla(0,0%,53%,.12)'}}>
            <ul className="menu-container">
                <li className="menu-item"> <FontAwesomeIcon icon={faStopwatch}></FontAwesomeIcon> <span>{texts.goldtagproduct}</span></li>
                <li className="menu-item"> <FontAwesomeIcon icon={faIdCard}></FontAwesomeIcon> <span>{texts.goldtagservice}</span></li>
                {texts.dropdownmenu.map((item)=>{
                    return(
                        <li className="menu-item">
                            <FontAwesomeIcon icon={item.icon}></FontAwesomeIcon>
                            <span>{item.title}</span>
                            <DropDown menu={item} />
                        </li>
                    )
                })}
                <li className="menu-item"><FontAwesomeIcon icon={faUser}></FontAwesomeIcon>{texts.login}</li>
                <li className="menu-item"><FontAwesomeIcon icon={faUserNurse}></FontAwesomeIcon>{texts.signup}</li>
            </ul>
            <hr></hr>
        </div>
    )
}
export default Navbar;