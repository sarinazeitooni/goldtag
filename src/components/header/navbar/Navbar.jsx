import './style/navbar.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import DropDown from "./dropdown/Dropdown";
import navbarData from './Data/navbarData';
import {faStopwatch, faIdCard, faUser, faUserNurse}
from "@fortawesome/free-solid-svg-icons";
function Navbar(){
    return(
        <div id="navbar" className='sticky'>
            <ul className="menu-container">
                <li className="menu-item"> <FontAwesomeIcon icon={faStopwatch}></FontAwesomeIcon> <span>{navbarData.goldtagproduct}</span></li>
                <li className="menu-item"> <FontAwesomeIcon icon={faIdCard}></FontAwesomeIcon> <span>{navbarData.goldtagservice}</span></li>
                {navbarData.dropdownmenu.map((item)=>{
                    return(
                        <li className="menu-item">
                            <FontAwesomeIcon icon={item.icon}></FontAwesomeIcon>
                            <span>{item.title}</span>
                            <DropDown menu={item} />
                        </li>
                    )
                })}
                <li className="menu-item"><FontAwesomeIcon icon={faUser}></FontAwesomeIcon>{navbarData.login}</li>
                <li className="menu-item"><FontAwesomeIcon icon={faUserNurse}></FontAwesomeIcon>{navbarData.signup}</li>
            </ul>
            <hr></hr>
        </div>
    )
}
export default Navbar;