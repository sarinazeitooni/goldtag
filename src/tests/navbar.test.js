import React from 'react';
import {shallow} from 'enzyme';
import DropDown from "../components/header/navbar/dropdown/Dropdown";
import Navbar from "../components/header/navbar/Navbar";
describe("Navbar component" , ()=>{
    it("should match snap shot" , ()=>{
        const wrapper = shallow(<Navbar/>);
        expect(wrapper).toMatchSnapshot();
    });
    it("should render container" , ()=>{
        const wrapper = shallow(<Navbar/>);
        const container = wrapper.find('ul.menu-container');
        expect(container.exists()).toBe(true);
    });
    it("should render drop down" , ()=>{
        const wrapper = shallow(<Navbar/>);
        const dropDown = wrapper.find('li.menu-item');
        expect(dropDown.exists()).toBe(true);
    });
})