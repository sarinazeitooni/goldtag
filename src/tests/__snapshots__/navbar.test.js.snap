// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should match snap shot 1`] = `ShallowWrapper {}`;
